### ISSUE TITLE
Content-Security-Policy errors

### ISSUE THREAD

**Issue by jckonczal** [validator/validator#2062]
### Summary

The HTML checker reports errors about images, scripts, CSS files, and other things from the same web server when there is a Content-Security-Policy that includes a default-src directive that allows 'self'.  For example, look at the report for https://usamricd.health.mil.  I have allowed every source necessary to make the website work in various browsers, but there are 20 warnings that files are blocked.  This also happens with https://cccrp.health.mil.

### URL for document that reproduces the problem

https://cccrp.health.mil

### HTML/CSS/SVG snippet that reproduces the problem

```html
<meta http-equiv="Content-Security-Policy" content="default-src 'self'; img-src 'self' https://mrdc.health.mil">
...
<link href="/assets/css/cccrp-styles.css" rel="stylesheet">
<link href="/assets/css/us-official.css" rel="stylesheet">
<link href="/assets/css/lapse.css" rel="stylesheet">
...
<img src="/assets/images/home/cccrp-logo.png" class="logoimage-2nd" alt="">
...
<script src="/assets/js/bootstrap.bundle.min.js"></script>
<script src="/assets/js/us-official.js"></script>
```

### Checker error messages

```shell
Warning: Resource violates Content Security Policy (meta tag): external stylesheet /assets/css/cccrp-styles.css blocked by style-src directive.

From line 1, column 598; to line 1, column 656

ylesheet"><link href="/assets/css/cccrp-styles.css" rel="stylesheet"><link 

Warning: Resource violates Content Security Policy (meta tag): external stylesheet /assets/css/us-official.css blocked by style-src directive.

From line 1, column 657; to line 1, column 714

ylesheet"><link href="/assets/css/us-official.css" rel="stylesheet"><link 

Warning: Resource violates Content Security Policy (meta tag): external stylesheet /assets/css/lapse.css blocked by style-src directive.

From line 1, column 715; to line 1, column 766

ylesheet"><link href="/assets/css/lapse.css" rel="stylesheet"><link 

Warning: Resource violates Content Security Policy (meta tag): image /assets/images/home/cccrp-logo.png blocked by img-src directive.

From line 1, column 6358; to line 1, column 6432

logo-2nd"><img src="/assets/images/home/cccrp-logo.png" class="logoimage-2nd" alt=""></div>

...

Warning: Resource violates Content Security Policy (meta tag): external script /assets/js/bootstrap.bundle.min.js blocked by script-src directive.

From line 1, column 11849; to line 1, column 11897

></script><script src="/assets/js/bootstrap.bundle.min.js"></scri

Warning: Resource violates Content Security Policy (meta tag): external script /assets/js/us-official.js blocked by script-src directive.

From line 1, column 11907; to line 1, column 11946

></script><script src="/assets/js/us-official.js"></scri
```

### Steps to reproduce

1.
2.
3.

### Expected behavior

No warnings about stylesheets from 'self', allowed by default-src, being blocked by style-src, and images blocked by img-src, etc.

### Actual behavior

The checker warns that stylesheets, images, etc., allowed in Content-Security-Policy by "default-src 'self'", are blocked.  Browsers' correct rendering of the pages show that those things are not blocked.

**Comment by sideshowbarker** [comment]
Thanks much for catching this and for taking time to report it — and for providing a good repro case.

Fixed in https://github.com/validator/validator/commit/142931395412c00434ffb40a14d65992efd17aa8 and deployed: https://validator.w3.org/nu/?doc=https://cccrp.health.mil

**Comment by jckonczal** [comment]
Thank you for fixing this so quickly.  However, it looks like it's still broken when I upload my

[index.html](https://github.com/user-attachments/files/26416658/index.html)

 HTML file instead of checking the live website.

**Comment by sideshowbarker** [comment]
> Thank you for fixing this so quickly.  However, it looks like it's still broken when I upload my
> 
> [index.html](https://github.com/user-attachments/files/26416658/index.html)
> 
>  HTML file instead of checking the live website.

That’s expected. An uploaded file doesn’t have an origin. So a `'self'` check in that case will never match anything. The CSP requirements for `'self'` necessitate a document having an associated origin — being deployed/served from a scheme+host+port URL. So CSP enforcement of `'self'` checking also requires a real origin to check against.

**Comment by jckonczal** [comment]
That's not an unreasonable expectation, given your explanation, but it was surprizing to me.  I understand that you wouldn't be able to check resources specified with full URLs against the origin.  By assuming a hypothetical origin, however, it would be posslble to check relative URLs, which would be convenient.
